routing/pathfind.go

```diff
@@ -159,6 +194,16 @@ func newRoute(sourceVertex route.Vertex,
 			totalTimeLock += uint32(finalHop.cltvDelta)
 
 			outgoingTimeLock = currentHeight + uint32(finalHop.cltvDelta)
+
+			// If we have a feature vector for the final hop, we'll
+			// determine if they support TLV onion payloads by
+			// checking if a var_onion_optin feature is set.
+			// Regardless of what the graph suggests, if the invoice
+			// has feature bits we will assume those override
+			// anything on a node announcement.
+			tlvPayload = supports(
+				finalHop.features, lnwire.TLVOnionPayloadOptional,
+			)
```

Comment:
Separation of concerns doesn't feel completely right here. Is `pathEdges []*channeldb.ChannelEdgePolicy` still the right thing to pass into this function? A different structure that contains exactly what is needed could maybe remove the need to do this interpretation here.

Reply:
indeed i think there is a more minimal struct that would suffice, ideally tied to our final design for the router cache